modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/JaxbWorkflowOperationInstance.java

```diff
@@ -0,0 +1,226 @@
+      return v == null ? null : v.getTime();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see javax.xml.bind.annotation.adapters.XmlAdapter#unmarshal(java.lang.Object)
+     */
+    @Override
+    public Date unmarshal(Long v) throws Exception {
+      return v == null ? null : new Date(v);
+    }
+  }
+
+  @XmlAttribute(name = "id")
+  private Long id;
+
+  @XmlAttribute(name = "template")
+  protected String template;
+
+  @XmlAttribute(name = "job")
+  protected Long jobId;
+
+  @XmlAttribute(name = "state")
+  protected WorkflowOperationInstance.OperationState state;
+
+  @XmlAttribute(name = "description")
+  protected String description;
+
+  @XmlElement(name = "configuration")
+  @XmlElementWrapper(name = "configurations")
+  protected Set<JaxbWorkflowConfiguration> configurations;
+
+  @XmlElement(name = "holdurl")
+  protected String holdStateUserInterfaceUrl;
+
+  @XmlElement(name = "hold-action-title")
+  protected String holdActionTitle;
+
+  @XmlAttribute(name = "fail-on-error")
+  protected boolean failWorkflowOnException;
+
+  @XmlAttribute(name = "if")
+  protected String executeCondition;
+
+  @XmlAttribute(name = "unless")
+  protected String skipCondition;
+
+  @XmlAttribute(name = "exception-handler-workflow")
+  protected String exceptionHandlingWorkflow;
+
+  @XmlAttribute(name = "abortable")
+  protected Boolean abortable;
+
+  @XmlAttribute(name = "continuable")
+  protected Boolean continuable;
+
+  @XmlJavaTypeAdapter(JaxbWorkflowOperationInstance.DateAdapter.class)
+  @XmlElement(name = "started")
+  protected Date dateStarted;
+
+  @XmlJavaTypeAdapter(JaxbWorkflowOperationInstance.DateAdapter.class)
+  @XmlElement(name = "completed")
+  protected Date dateCompleted;
+
+  @XmlElement(name = "time-in-queue")
+  protected Long timeInQueue;
+
+  @XmlAttribute(name = "max-attempts")
+  protected int maxAttempts;
+
+  @XmlAttribute(name = "failed-attempts")
+  protected int failedAttempts;
+
+  @XmlAttribute(name = "execution-host")
+  protected String executionHost;
+
+  @XmlJavaTypeAdapter(RetryStrategy.Adapter.class)
+  @XmlAttribute(name = "retry-strategy")
+  protected RetryStrategy retryStrategy;
+
+  /**
+   * No-arg constructor needed for JAXB serialization
+   */
+  public JaxbWorkflowOperationInstance() {
+    this.maxAttempts = 1;
+    this.retryStrategy = RetryStrategy.NONE;
+  }
+
+  /**
+   * Builds a new workflow operation instance based on another workflow operation.
+   *
+   */
+  public JaxbWorkflowOperationInstance(WorkflowOperationInstance operation) {
+    this();
+    this.id = operation.getId();
+    this.template = operation.getTemplate();
+    this.jobId = operation.getId();
+    this.state = operation.getState();
+    this.description = operation.getDescription();
+    if (operation.getConfigurations() != null) {
+      this.configurations = Optional.ofNullable(operation.getConfigurations().entrySet())
+              .orElseGet(Collections::emptySet)
```

Comment:
`entrySet` should never return `null`:

```suggestion
      this.configurations = operation.getConfigurations().entrySet()
```